src/cli.rs

```diff
@@ -0,0 +1,236 @@
+        let config_path = config_dir.join("inlyne").join("inlyne.toml");
+        if config_path.is_file() {
+            let text = fs::read_to_string(&config_path).context("Failed reading config file")?;
+            let config = toml::from_str(&text)?;
+            Ok(config)
+        } else {
+            Ok(Self::default())
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Args {
+    pub file_path: PathBuf,
+    pub theme: color::Theme,
+    pub scale: Option<f32>,
+}
+
+fn command<'a>(scale_help: &'a str, default_theme: ThemeType) -> Command<'a> {
+    let file_arg = Arg::new("file")
+        .required(true)
+        .takes_value(true)
+        .value_name("FILE")
+        .value_parser(value_parser!(PathBuf))
+        .help("Path to the markdown file");
+    let theme_arg = Arg::new("theme")
+        .short('t')
+        .long("theme")
+        .takes_value(true)
+        .value_parser(value_parser!(ThemeType))
+        .default_value(default_theme.as_str())
+        .help("Theme to use when rendering");
+
+    let scale_arg = Arg::new("scale")
+        .short('s')
+        .long("scale")
+        .takes_value(true)
+        .value_parser(value_parser!(f32))
+        .help(scale_help);
+
+    command!().arg(file_arg).arg(theme_arg).arg(scale_arg)
+}
+
+impl Args {
+    pub fn parse() -> Self {
+        let args = env::args_os().collect();
+        let config = match Config::load() {
+            Ok(config) => config,
+            Err(err) => {
+                // TODO: switch to logging
+                eprintln!(
+                    "WARN: Failed reading config file. Falling back to defaults. Error: {}",
+                    err
+                );
+                Config::default()
+            }
+        };
+
+        Self::parse_from(args, config)
+    }
+
+    fn parse_from(args: Vec<OsString>, config: Config) -> Self {
+        let scale_help = format!(
+            "Factor to scale rendered file by [default: {}]",
+            match config.scale {
+                Some(scale) => scale.to_string(),
+                None => String::from("Window's scale factor"),
+            }
+        );
+
+        let command = command(&scale_help, config.theme.unwrap_or_default());
+        let matches = command.get_matches_from(args);
+
+        let file_path = matches.get_one("file").cloned().expect("required");
+        let theme = matches
+            .get_one::<ThemeType>("theme")
+            .cloned()
+            .unwrap_or_default()
+            .as_theme();
+        let scale = matches.get_one("scale").cloned().or(config.scale);
+
+        Self {
+            file_path,
+            theme,
+            scale,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn gen_args(args: Vec<&str>) -> Vec<OsString> {
+        std::iter::once("inlyne")
+            .chain(args.into_iter())
+            .map(OsString::from)
+            .collect()
+    }
+
+    #[test]
+    fn debug_assert() {
```

Comment:
What is the purpose of the same test twice with a different scaling help text. Especially since "1.5" or "window's scale factor" isn't what the program would use at runtime for the help text.

Reply:
It doesn't really make sense to have that test twice, I agree

I switched it to just one test using the same help message that would be passed in (even though the specific contents won't have an impact)